### ISSUE TITLE
Sidebar regression since version 0.11.1-1

### ISSUE THREAD

**Issue by WMTGCF** [typora/typora-issues#4570]
**High CPU load and lagging behavior**

Since version 0.11.1-1 the display and response of the sidebar with many folders and files is very slow and the application is unresponsive. The CPU usage is high (1 core/thread) is fully loaded.

**To Reproduce**
1. Install version 0.10.11-1 - everything is working fine
2. Upgrade to version 0.11.1-1 (or 0.11.2-1) - lagging sidebar and application is unresponsive
3. Downgrade to version 0.10.11-1 - everything ist working fine again

**Desktop**
* Ubuntu 20.04 focal
* Kernel 5.4.0-80
* CPU Intel Core i7-9700T
* RAM 64 GB
* SDD

**Typora Version**
0.11.1-1 and 0.11.2-1

**Comment by abnerlee** [comment]
Possible to attach a screencast?

**Comment by WMTGCF** [comment]
Screencasts are attached for version 0.10.11 and 0.11.2. (Unfortunately, I have no way to make the mouse clicks visible. I also stopped recording for 0.11.2 because the waiting time is way too long).

https://user-images.githubusercontent.com/88326277/128143711-acc245de-489a-4007-8cec-9c9809c812c6.mp4

https://user-images.githubusercontent.com/88326277/128143756-6592a782-94cc-4e49-834f-082437405543.mp4

For the screencasts, I created a stripped down version of the directory structure with the following commands:

```bash
eval "mkdir -p ./{A..Z}/{a..z}/{A..Z}/"
eval "touch ./{A..Z}/{a..z}/{A..Z}/content{1..3}.md"
```

In my production system I have many times more directories and files (md, pdf, doc, ...).

As far as I have noticed, since version 0.11.1 all files are "scanned" or "indexed" after startup. If you wait long enough after startup at the created directory structure, typora reacts normally again. This should explain the sluggish behavior.

But since I have a lot of directories and files in my production system, I can't wait "for hours" every time after startup until I can work with typora.
